core/server/services/mega/advertising.js

```diff
@@ -0,0 +1,119 @@
+                    <a class="c-sponsorpromo__link" href="https://ad.doubleclick.net/ddm/clk/507046894;302742333;f" style="-webkit-text-size-adjust: 100%; -ms-text-size-adjust: 100%; color: currentColor; text-decoration: none;">
+                      Sponsor Content by ServiceNow
+                    </a>
+                  </h3>
+                </td>
+              </tr>
+
+
+
+              <tr>
+                <td align="left" valign="top" width="100%" style="-webkit-text-size-adjust: 100%; -ms-text-size-adjust: 100%; mso-table-lspace: 0pt; mso-table-rspace: 0pt;">
+                  <h4 class="c-sponsorpromo__title" style="color: #21272c; font-family: 'Graphik', -apple-system, BlinkMacSystemFont, 'Google Sans', 'Roboto', 'Helvetica Neue', Helvetica, Segoe UI, Arial, sans-serif; font-weight: 600; font-size: 16px; line-height: 24px; margin: 0 0 8px 0;">
+                    <a class="c-sponsorpromo__link" href="https://ad.doubleclick.net/ddm/clk/507046894;302742333;f" style="-webkit-text-size-adjust: 100%; -ms-text-size-adjust: 100%; color: currentColor; text-decoration: none;">
+                      How Do We Recover?
+                    </a>
+                  </h4>
+                </td>
+              </tr>
+
+
+              <tr>
+                <td align="left" valign="top" width="100%" class="c-sponsorpromo__content" style="-webkit-text-size-adjust: 100%; -ms-text-size-adjust: 100%; mso-table-lspace: 0pt; mso-table-rspace: 0pt; padding-bottom: 24px;">
+                  <p class="c-sponsorpromo__dek" style="color: #21272c; font-family: 'Graphik', -apple-system, BlinkMacSystemFont, 'Google Sans', 'Roboto', 'Helvetica Neue', Helvetica, Segoe UI, Arial, sans-serif; font-weight: 400; font-size: 14px; line-height: 24px; margin: 0;">
+                    <a class="c-sponsorpromo__link" href="https://ad.doubleclick.net/ddm/clk/507046894;302742333;f" style="-webkit-text-size-adjust: 100%; -ms-text-size-adjust: 100%; color: currentColor; text-decoration: none;">
+                      Real recovery requires more than just vaccinating the masses and distributing checks. <b style="color: #21272c; text-decoration: underline; text-decoration-color: #655ade; font-weight: normal;">Find out how digital workflows can help bring our communities relief—and ensure the holistic recovery we need.</b>&nbsp;
+                    </a>
+                  </p>
+                </td>
+              </tr>
+
+            </tbody>
+          </table>
+        </td>
+      </tr>
+    </tbody>
+  </table>
+
+  ${markers.other}
+`
+
+
+
+/**
+ * The Atlantic's custom module inserting
+ * ads into newsletters.
+ *
+ * Aim to keep the surface area of the fork entirely in this file.
+ *
+ * @param {Object}
+ * @property {Any} site
+ * @property {String} HTML
+ * @returns String
+ *
+ */
+async function fillAdvertisements({site, html}) {
+
+  // Title with all non-word characters stripped out
+  const siteID = site.title.replace(/[\W_]+/g,"").toLowerCase();
```

Comment:
Using site title seems risky since the author could change this for legitimate reasons. URL might be more reliable? (It'll be tricky in development since the value would always be `http://localhost:2368/` but in production the URL should be unique and unchangeable)

Reply:
I had the same thought and was planning to revisit it now that it works. For development and testing we `AD_SITE_ID` could be an ENV var or something, allowing us to test against real Hattie instances.

Reply:
You could also do something like this to remove the risk of trailing slash, port, or protocol weirdness:

```
const siteURL = new URL(site.url);
const siteID = encodeURIComponent(siteURL.hostname);
```

Reply:
True. The weirdness is kind of useful for local development to prevent it from being just "localhost." I'll put this on my list to revisit once we have a little more info about how these are being hosted.